lib/slackify/handlers/validator.rb

```diff
@@ -19,10 +19,28 @@ def self.verify_handler_integrity(handler)
         handler.dig(handler_name, 'commands').each do |command|
           command_errors = []
 
-          unless command['regex'].is_a?(Regexp)
+          unless command['regex'] || command['base_command']
+            command_errors.append('No regex or base command was provided.')
+          end
+
+          if command['regex'].present?
+            if command['base_command'].present?
+              command_errors.append('Regex and base_command cannot be used in the same handler.')
+            end
+
+            if command['parameters'].present?
+              command_errors.append('Regex and parameters cannot be used in the same handler.')
+            end
+          end
+
+          unless command['regex'].nil? || command['regex'].is_a?(Regexp)
```

Comment:
shouldn't is be `command['regex'] && command['regex'].is_a?(Regexp)`? We wouldn't enter the block if the regex is nil

Reply:
This one I moved to the block above to make it easier to read and removed the `command['regex'].nil?`